core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -66,26 +67,54 @@ oppia.directive('topNavigationBar', [
           $scope.onLogoutButtonClicked = function() {
             $window.localStorage.removeItem('last_uploaded_audio_lang');
           };
-
-          $scope.profileDropdownIsActive = false;
-          $scope.onMouseoverProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
-            $scope.profileDropdownIsActive = true;
-          };
-          $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
-            $scope.profileDropdownIsActive = false;
+          $scope.openSubmenu = function (e, menu) {
+            angular.element('.nav a').blur();
+            angular.element(e.currentTarget).focus();
+            $scope.activeMenu = menu;
           };
-          $scope.onMouseoutDropdownMenuAbout = function(evt) {
-            angular.element(evt.currentTarget)[0].blur();
+          $scope.closeSubmenu = function (evt) {
+            $scope.activeMenu = '';
+            if (evt.currentTarget.nodeName === 'UL') {
+              if (angular.element(evt.currentTarget).closest('li')
+                .find('a').is(':focus')) {
+                angular.element(evt.currentTarget).closest('li')
+                  .find('a').blur();
+              }
+            }
           };
-          $scope.onMouseoverDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
+          $scope.isEventOccured = function (e, targetEvent) {
+            if (targetEvent === 'enter' && e.keyCode === 13) {
+              return true;
+            } else if (targetEvent === 'tab' && e.keyCode === 9) {
+              return true;
+            } else if (targetEvent === 'shiftTab' &&
+              (e.shiftKey && e.keyCode === 9)) {
+              return true;
+            } else {
+              return false;
+            }
           };
-          $scope.onMouseoutDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
+          /*
+          @param e - Event occured.
+          @param name - menu(aboutMenu,profileMenu) to open/close.
+          @param targetEvent - Event to perform the action(Enter,Tab,Shift+Tab)
+          @param action - Action to be performed(open/close).
+          */
+          $scope.handleNavigation = function (e, menu, targetEvent, action) {
+            if ($scope.isEventOccured(e, targetEvent)) {
+              if (action === 'open') {
+                $scope.openSubmenu(e, menu);
+              } else {
+                $scope.closeSubmenu(e);
+              }
+            }
           };
 
+          angular.element(document).on('click', function (e) {
```

Comment:
Maybe add comment explaining what this is for.

Also -- does this negate clicking on the menu itself?